### ISSUE TITLE
[FEA] Add more functionality to `cudf.io.read_parquet_metadata` API

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#11214]
**Is your feature request related to a problem? Please describe.**
It would be nicer to have the row-group wise metadata returned instead of returning just the number of row-groups. That way users can identify how many rows & columns are stored in each row-group.

**Comment by galipremsagar** [comment]
@vuule Did you intended to the same pyarrow parquet schema? Like:
```python
(Pdb) x = pq.ParquetFile(fname)
(Pdb) x.metadata
<pyarrow._parquet.FileMetaData object at 0x7fbe0a172040>
  created_by: parquet-cpp-arrow version 8.0.0
  num_columns: 15
  num_rows: 0
  num_row_groups: 1
  format_version: 1.0
  serialized_size: 7481
(Pdb) x.metadata.row_group
<built-in method row_group of pyarrow._parquet.FileMetaData object at 0x7fbe0a172040>
(Pdb) x.metadata.row_group(0)
<pyarrow._parquet.RowGroupMetaData object at 0x7fbe0a172270>
  num_columns: 15
  num_rows: 0
  total_byte_size: 196
```

or do we want to extract only the necessary bit of details and return those?

**Comment by vuule** [comment]
I like the option to follow the pyarrow metada structure here, if it's not a huge overhead to gather.

**Comment by galipremsagar** [comment]
Yea, should not be an issue since we already tap into this API anyways: https://github.com/rapidsai/cudf/blob/branch-22.08/python/cudf/cudf/io/parquet.py#L199

**Comment by GregoryKimball** [comment]
> That way users can identify how many rows & columns are stored in each row-group.

Now that we have `read_parquet_metadata` from #13663, could we re-scope this issue to specify the changes we would like to see in the `parquet_metadata` class?

As I understand, the number of columns will be identical for all row groups. We could add row counts for each row group as a new vector, or perhaps to `metadata`. Where are the row group min/max statistics stored?

**Comment by mhaseeb123** [comment]
Closed by #15398
